src/main/java/com/frostwire/jlibtorrent/TorrentStats.java

```diff
@@ -0,0 +1,150 @@
+package com.frostwire.jlibtorrent;
+
+import com.frostwire.jlibtorrent.alerts.Alert;
+import com.frostwire.jlibtorrent.alerts.AlertType;
+import com.frostwire.jlibtorrent.alerts.TorrentAlert;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * It will start an alertEvent listener to keep track of the download
+ * and upload Speed (bytes/sec). The granularity selected will depend
+ * on the speed in which the results of the speed of the Alert event
+ * listener is retrieved, which can have a delay in milliseconds.
+ *
+ * @author haperlot
+ */
+
+public final class TorrentStats {
+
+    private final SessionManager sessionManager;
+    private final TorrentHandle torrentHandle;
+    private final long samplingIntervalInMs;
+    private final int[] STATS = {AlertType.STATS.swig()};
+    private final long MAX_SAMPLES;
+    private final Queue<Integer> downloadRate = new LinkedList<>();
+    private final Queue<Integer> uploadRate = new LinkedList<>();
+    private long tStart; //for collecting sampling interval time
+
+    public enum Metric {
+        UploadRate,
+        DownloadRate
+    }
+
+    public TorrentStats(final SessionManager sessionManager, TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        this.sessionManager = sessionManager;
+        this.torrentHandle = torrentHandle;
+        this.samplingIntervalInMs = samplingIntervalInMs;
+        this.tStart = System.currentTimeMillis();
+        this.MAX_SAMPLES = maxHistoryInMs / samplingIntervalInMs;
+        startAlertListener();
+    }
+
+    /**
+     * It will start the alert event listener, as soon as the sampling interval
+     * meets the requested time it will save the values on queues. If the queues
+     * reach max size denoted by the maxHistory time, they will poll() the head
+     * element, in our case the oldest inserted.
+     */
+    public void startAlertListener() {
+        sessionManager.addListener(new AlertListener() {
+            @Override
+            public int[] types() {
+                return STATS;
+            }
+
+            @Override
+            public void alert(Alert<?> alert) {
+
+                if (!(alert instanceof TorrentAlert<?>)) {
```

Comment:
this check should be unnecessary, this should never happen because you've already stated the only stats coming in will be `stat_alert`, which extend from `torrent_alert` in C++-land